android/src/main/java/ee/forgr/audio/NativeAudio.java

```diff
@@ -255,6 +283,276 @@ public void run() {
+                 * optionally deletes the source file if it resides in an application-safe directory. If `autoPlay` is true and
+                 * notifications are enabled, the media notification is updated to reflect playback.
+                 *
+                 * <p>On success the method resolves the PluginCall with an object containing the generated `assetId`. On failure the
+                 * PluginCall is rejected with a descriptive error message.
+                 */
+                @Override
+                public void run() {
+                    try {
+                        plugin.initSoundPool();
+
+                        // Generate unique temporary asset ID
+                        final String assetId =
+                            "playOnce_" + System.currentTimeMillis() + "_" + UUID.randomUUID().toString().substring(0, 8);
+
+                        // Extract options
+                        String assetPath = call.getString(ASSET_PATH);
+                        if (!plugin.isStringValid(assetPath)) {
+                            call.reject(ERROR_ASSET_PATH_MISSING);
+                            return;
+                        }
+
+                        boolean autoPlay = call.getBoolean("autoPlay", true);
+                        final boolean deleteAfterPlay = call.getBoolean("deleteAfterPlay", false);
+                        float volume = call.getFloat(VOLUME, 1F);
+                        boolean isLocalUrl = call.getBoolean("isUrl", false);
+                        int audioChannelNum = 1; // Single channel for playOnce
+
+                        // Track this as a playOnce asset
+                        plugin.playOnceAssets.add(assetId);
+
+                        // Store notification metadata if provided
+                        JSObject metadata = call.getObject(NOTIFICATION_METADATA);
+                        if (metadata != null) {
+                            Map<String, String> metadataMap = new HashMap<>();
+                            if (metadata.has("title")) metadataMap.put("title", metadata.getString("title"));
+                            if (metadata.has("artist")) metadataMap.put("artist", metadata.getString("artist"));
+                            if (metadata.has("album")) metadataMap.put("album", metadata.getString("album"));
+                            if (metadata.has("artworkUrl")) metadataMap.put("artworkUrl", metadata.getString("artworkUrl"));
+                            if (!metadataMap.isEmpty()) {
+                                plugin.notificationMetadataMap.put(assetId, metadataMap);
+                            }
+                        }
+
+                        // Preload the asset using the helper method
+                        try {
+                            // Check if asset already exists
+                            if (plugin.audioAssetList.containsKey(assetId)) {
+                                throw new Exception(ERROR_AUDIO_EXISTS + " - " + assetId);
```

Comment:
This looks like an antipattern to me.

Throwing -> catching and doing `call.reject(ex.getMessage());` makes no sense. You should reject and not throw. This double error handling has no purpose, and it makes it harder to see that you are actually rejecting the call.

```suggestion
call.reject(ERROR_AUDIO_EXISTS + " - " + assetId);
```